Build failing on macOS 11.3

I tried to build lite-xl from source using the instructions given but a problem kept occurring talking about `Undefined symbols`. So I went digging (after I triple checked that I have all dependancies installed).

394f980 was the first to compile and 032018e also seemed to work. But f913a85 introduced something breaking. 53f77a2 resolved the issue but still left an other one behind. The one about the mysterious "Undefined symbols".

Now I'm no `objective-c` developer and I have no clue whats going on when looking at the code but it seems a lot like there's something fishy going on with the `Foundation` framework specifically `NSBundle` (in `bundle_open.m`).

Following is there error I get from 53f77a2 and onwards.

```
[46/46] Linking target src/lite
FAILED: src/lite
c++  -o src/lite src/lite.p/api_api.c.o src/lite.p/api_cp_replace.c.o src/lite.p/api_renderer.c.o src/lite.p/api_renderer_font.c.o src/lite.p/api_system.c.o src/lite.p/renderer.c.o src/lite.p/rencache.c.o src/lite.p/main.c.o src/lite.p/bundle_open.m.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -Wl,-rpath,/usr/local/opt/freetype/lib -Wl,-rpath,/usr/local/Cellar/libx11/1.7.1/lib lib/font_renderer/libfontrenderer.a subprojects/lua/src/liblua.a -lm -ldl -L/usr/local/lib -lSDL2 /usr/local/Cellar/libx11/1.7.1/lib/libX11.dylib -Wl,-rpath,/usr/local/Cellar/libagg/2.5/lib /usr/local/Cellar/libagg/2.5/lib/libagg.dylib /usr/local/opt/freetype/lib/libfreetype.dylib
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_NSBundle", referenced from:
      objc-class-ref in bundle_open.m.o
  "_objc_autoreleasePoolPop", referenced from:
      _set_macos_bundle_resources in bundle_open.m.o
  "_objc_autoreleasePoolPush", referenced from:
      _set_macos_bundle_resources in bundle_open.m.o
  "_objc_msgSend", referenced from:
      _set_macos_bundle_resources in bundle_open.m.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

Any help is greatly appreciated!

--- a/meson.build
+++ b/meson.build
@@ -33,6 +33,9 @@ lite_link_args = []
 if cc.get_id() == 'gcc' and get_option('buildtype') == 'release'
     lite_link_args += ['-static-libgcc', '-static-libstdc++']
 endif
+if host_machine.system() == 'darwin'
+    lite_link_args += ['-framework', 'CoreSerivces', '-framework', 'Foundation']
+endif
 
 lite_rc = []
 if host_machine.system() == 'windows'

--- a/meson.build
+++ b/meson.build
@@ -34,7 +34,7 @@ if cc.get_id() == 'gcc' and get_option('buildtype') == 'release'
     lite_link_args += ['-static-libgcc', '-static-libstdc++']
 endif
 if host_machine.system() == 'darwin'
-    lite_link_args += ['-framework', 'CoreSerivces', '-framework', 'Foundation']
+    lite_link_args += ['-framework', 'CoreServices', '-framework', 'Foundation']
 endif
 
 lite_rc = []

